src/main/java/org/corpus_tools/pepper_WebannoTSVModule/WebannoTSV2SaltMapper.java

```diff
@@ -387,43 +399,49 @@ public DOCUMENT_STATUS mapSDocument() {
 																													// target
 																													// node
 																													// ID
-												String[] position_source_target = edgeSourceTarget.split("\\[");
-												position = position_source_target[0];
-												String[] source_target = position_source_target[1].split("_");
-												source = source_target[0];
-												target = source_target[1].substring(0, source_target[1].length() - 1);
-											} else { // this is a link between
-														// two single token
-														// spans, just use token
-														// IDs
-												position = edgeSourceTarget;
-											}
-											if (target.equals("0")) {
+												String[] position_source_target = edgeSourceTarget.split("\\[");							
+//												position = position_source_target[0];
+//												String[] source_target = position_source_target[1].split("_");
+//												source = source_target[0];
+//												target = source_target[1].substring(0, source_target[1].length() - 1);
+												
+												String sourceTok = position_source_target[0];
+												String[] source_target_anno_ids = position_source_target[1].split("_");
+												String sourceTokAnnoId= source_target_anno_ids[0];
+												String targetTokAnnoId= source_target_anno_ids[1].substring(0, source_target_anno_ids[1].length() - 1);
+												
+												if(!sourceTokAnnoId.equals("0")) {
+													source = sourceTok+"["+sourceTokAnnoId+"]";
+												}else {
+													source = tokID;
```

Comment:
This should probably be 

```suggestion
													source = sourceTok;
```
because the the span node disambiguation always has the format "[source_target]" and here the source node disambiguation ID is omitted. (https://inception-project.github.io/releases/37.5/docs/user-guide.html#_relation_annotations). This would also be coherent with the `if` block.

Also, it would be more clear what is meant by switching the if-else condition:

```java
if(sourceTokAnnoId.equals("0") {
// ...
} else {
// ..
}
```